master/buildbot/www/auth.py

```diff
@@ -113,17 +113,20 @@ def __init__(self, header=None, headerRegex=None, **kwargs):
         if header is not None:
             self.header = header
         if headerRegex is not None:
-            self.headerRegex = re.compile(unicode2bytes(headerRegex))
+            self.headerRegex = re.compile(headerRegex)
 
     @defer.inlineCallbacks
     def maybeAutoLogin(self, request):
         header = request.getHeader(self.header)
         if header is None:
-            raise Error(403, b"missing http header " + self.header + b". Check your reverse proxy config!")
+            err = "missing http header " + self.header + \
+                  ". Check your reverse proxy config!"
+            raise Error(403, err.encode('utf-8'))
```

Comment:
`unicode2bytes()` helper function should be used instead of bare `.encode('utf-8')`

Reply:
You're right.

I made a lot of mistakes, I wish I could run the tests on my machine.